api/config-dev.yaml

```diff
@@ -1,10 +1,42 @@
+BatchRunnerConfig:
+  TimeInterval: 10s
 DbConfig:
   User: turing
   Password: turing
 DeployConfig:
   EnvironmentType: dev 
 EnsemblingJobConfig:
   DefaultEnvironment: dev
+  BatchSize: 10
+  MaxRetryCount: 3
+  ImageConfig:
+    Registry: ghcr.io
+    BaseImageRef: ghcr.io/gojek/turing/batch-ensembler:0.0.0-build.1-98b071d
+    BuildNamespace: default
+    BuildContextURI: git://github.com/gojek/turing.git#refs/heads/main
+    DockerfileFilePath: engines/batch-ensembler/app.Dockerfile
+    BuildTimeoutDuration: 20m
+  KanikoConfig:
+    Image: gcr.io/kaniko-project/executor
+    ImageVersion: v1.5.2
+    ResourceRequestsLimits:
+      Requests:
+        CPU: "1"
+        Memory: 1Gi
+      Limits:
+        CPU: "1"
+        Memory: 1Gi
+SparkInfraConfig:
+  CorePerCPURequest: 1.5
```

Comment:
Ok, the Merlin authors wrote this so I took it from there (https://github.com/gojek/merlin/blob/main/api/batch/resource.go#L289-L296). But I think I understand the rationale behind this.

Setting CPU limits doesnt tell the container how many CPU it has, just CPU shares which is not known by the container itself. As far as the container knows it has the same number of cores as the machine it is being run on. i.e if the Kubernetes Node has 4 VCPUs, it will just say that the container has 4 VCPU, as proved by calling `cat /proc/cpuinfo | grep processor` (should see 4 lines).

By default, Spark will take all the cores in the machine that it is presented: https://spark.apache.org/docs/2.4.0/spark-standalone.html#starting-a-cluster-manually. Spark doesn't seem to understand containers so I'm guessing it will just take the number of CPUs in the machine.

So Merlin approximates the number of CPUs required via this ratio. Given the Kubernetes CPU request, it will approximate the number of CPUs required (in an integer because physical CPUs can't be divided).

Looking at the formula here: https://github.com/ashwinath/turing/blob/batch-ensembler-orchestration/api/turing/cluster/spark.go#L244-L251

So the base case will always be at least 1 CPU core. And this multiplier will try to match the number of CPU cores to the cpu limit of the container.